mitmproxy/net/udp.py

```diff
@@ -96,7 +126,8 @@ def __init__(
 
     def connection_made(self, transport: asyncio.BaseTransport) -> None:
         if self._transport is None:
-            self._transport = cast(asyncio.DatagramTransport, transport)
+            self._transport = cast(DatagramTransport, transport)
+            self._transport.set_protocol(self)
```

Comment:
Is setting `protocol` to `self` correct, though? Looking at the code in the Python `asyncio` module, the `Transport` and `Protocol` interfaces are distinct and serve different purposes.